include/deal.II/numerics/vector_tools.h

```diff
@@ -1381,6 +1381,105 @@ namespace VectorTools
+   * This function is an updated version of the project_boundary_values_curl_conforming
+   * function. The intention is to fix a problem when using the previous function in
+   * conjunction with non-rectangular geometries (i.e. elements with non-rectangular faces).
+   * The L2-projection method used has been taken from the paper "Electromagnetic scattering 
+   * simulation using an H (curl) conforming hp finite element method in three dimensions"
+   * by PD Ledger, K Morgan and O Hassan ( Int. J.  Num. Meth. Fluids, Volume 53, Issue 8, pages 1267–1296).
+   * 
+   * This function will compute constraints that correspond to Dirichlet boundary conditions of the form
+   * $\vec{n}\times\vec{E}=\vec{n}\times\vec{F}$
+   * i.e. the tangential components of $\vec{E}$ and $f$ shall coincide.
+   * 
+   * If the ConstraintMatrix @p constraints contained values or other
+   * constraints before, the new ones are added or the old ones overwritten,
+   * if a node of the boundary part to be used was already in the list of
+   * constraints. This is handled by using inhomogeneous constraints. Please
+   * note that when combining adaptive meshes and this kind of constraints, the
+   * Dirichlet conditions should be set first, and then completed by hanging
+   * node constraints, in order to make sure that the discretization remains
+   * consistent. See the discussion on conflicting constraints in the
+   * module on @ref constraints 
+   *
+   * This function is explicitly for use with FE_Nedelec elements, or with FESystem 
+   * elements which contain FE_Nedelec elements. It will throw an exception if called
+   * with any other finite element. The user must ensure that FESystem elements are
+   * correctly setup when using this function as this check not possible in this case.
+   * 
+   * The second argument of this function denotes the first vector component of the 
+   * finite element which corresponds to the vector function that you wish to constrain.
+   * For example, if we are solving Maxwell's equations in 3D and have components 
+   * $(E_x,E_y,E_z,B_x,B_y,B_z)$ and we want the boundary conditions
+   * $\vec{n}\times\vec{B}=\vec{n}\times\vec{f}$, then @p first_vector_component would
+   * be 3. The @p boundary_function must return 6 components in this example, with the first 3
+   * corresponding to $\vec{E}$ and the second 3 corresponding to $\vec{B}$.
+   * Vectors are implicitly assumed to have exactly <code>dim</code> components
+   * that are ordered in the same way as we usually order the coordinate directions,
+   * i.e. $x$-, $y$-, and finally $z$-component.
+   * 
+   * The parameter @p boundary_component corresponds to the number @p boundary_indicator
+   * of the face. numbers::internal_face_boundary_id is an illegal value, since it is 
+   * reserved for interior faces.
+   * 
+   * The last argument is denoted to compute the normal vector $\vec{n}$ at the
+   * boundary points.
+   * 
+   * <h4>Computing constraints</h4>
```

Comment:
Move this whole section up, under the place where you have the formula with the E=F. Then separate it from the discussion of the arguments to this function using
```
  <h4>Arguments to this function</h4>
```
or similar.